### ISSUE TITLE
UI freezes randomly with client error

### ISSUE THREAD

**Issue by heruan** [vaadin/flow#7399]
I'm on 14.1.5 and the UI often blocks after random actions (i.e. clicks, keypress). No error is shown on the server logs, but an error is shown in the client's console:

![image001 copy](https://user-images.githubusercontent.com/851745/72809876-d23b9780-3c5c-11ea-951f-e70db9a3af76.png)

The loading bar on the top never finishes and you need to refresh the page to continue. This happens randomly so I can't provide steps to reproduce, but there is heavy use of push updates from the server which might be involved.

**Comment by heruan** [comment]
Note: this happens after a while of no user action, but it's not deterministic.

**Comment by heruan** [comment]
Possibly related to #2210 and #4071.

**Comment by heruan** [comment]
A couple of more notes:
1. it happens in production mode and development mode both;
2. it happens running locally via Spring Boot and also deploying on a WildFly instance.

**Comment by mehdi-vaadin** [comment]
The same error used to occurr in Vaadin Portlets (https://github.com/vaadin/portlet/issues/166) and was fixed. But, this seems to have a different cause.

**Comment by Kivinaama** [comment]
We would seem to have the same issue. Also on 14.1.5, and with dev-build we can see the same client-side error. Happens also with production-build. Symptoms are the same. @TatuLund knows our case.

I think it could be somehow related to hearBeatInterval, as with default settings (heartBeatInterval=300) if I let the browser be idle for 20mins it happens, and with heartBeatInterval=60 if I idle for 5 minutes it happens.

I have had server-side logging for both UI Detach and VaadinSession Destroy and neither is triggered.

**Comment by Kivinaama** [comment]
Also, for us it happens both with closeIdleSession=true and closeIdleSession=false

**Comment by heruan** [comment]
This PR solves a similar issue in Framework: https://github.com/vaadin/framework/pull/11791

**Comment by heruan** [comment]
I have a case where when the user clicks a button, the listener triggers a server push which updates the UI: it might happen than the push reaches the client before the response to the click.

**Comment by heruan** [comment]
The PR which closed this effectively prevents the exception and the client error, but it does not solve the issue since the user action still fails (now without any message, as if it gets ignored) and the UI might still become unusable, e.g. if it happens clicking a link navigation won't work anymore.

**Comment by TatuLund** [comment]
@heruan This PR fixes the specific bug. I think in your case there is most likely another bug which requires more investigation. 

>  since the user action still fails

When resynchronization is done, user action may be lost. That is why there will be warning logged in server log. This is expected.

> if it happens clicking a link navigation won't work anymore.

This part requires more investigation, i.e. there is possibly a corner case that something odd happens when resynchronization is done during navigation. This is a new bug revealed after #7399 was fixed.

**Comment by heruan** [comment]
> When resynchronization is done, user action may be lost.

I understand, but this happens very frequently when there is a push from the server originated by a user action. Should I open a new issue?

**Comment by TatuLund** [comment]
> Should I open a new issue?

Yes that is what I meant, once we you have new data so that we can start digging deeper what is going on. Also I would recommend also verify the actual release once it is out.
